cumulusci/tasks/datadictionary.py

```diff
@@ -28,44 +54,139 @@ class GenerateDataDictionary(BaseGithubTask):
+                )
+
+                contents = repo.file_contents("cumulusci.yml", ref=ref)
+                cumulusci_yml = cci_safe_load(
+                    io.StringIO(contents.decoded.decode("utf-8"))
+                )
+                project = cumulusci_yml.get("project", {})
+                namespace = project.get("package", {}).get("namespace", "")
+                if namespace:
+                    namespace = f"{namespace}__"
+                else:
+                    namespace = ""
+
+                if f"{repo.owner}/{repo.name}" in visited:
+                    continue
+
+                deps.append(
+                    Package(
+                        repo,
+                        project.get("package", {}).get(
+                            "name", f"{repo.owner}/{repo.name}"
+                        ),
+                        namespace,
+                        project.get("git", {}).get("prefix_release", "release/"),
+                    )
+                )
+                visited.add(f"{repo.owner}/{repo.name}")
+
+                deps.extend(
+                    self._get_repo_dependencies(
+                        cumulusci_yml.get("project", {}).get("dependencies", []),
+                        visited_repos=visited,
+                    )
+                )
+
+        return deps
 
     def _run_task(self):
         self.logger.info("Starting data dictionary generation")
 
         self._init_schema()
-        self._walk_releases()
+
+        # Find all of our dependencies, if we're processing dependencies.
+        dependencies = self.project_config.project__dependencies or []
+        namespace = self.project_config.project__package__namespace
+        if namespace:
+            namespace = f"{namespace}__"
+        repos = [
+            Package(
+                self.get_repo(),
+                self.project_config.project__package__name,
+                namespace,
+                self.project_config.project__git__prefix_release,
+            )
+        ]
+        if "additional_dependencies" in self.options:
+            repos += self._get_repo_dependencies(
+                [
+                    {"github": url}
+                    for url in process_list_arg(self.options["additional_dependencies"])
+                ],
+                include_beta=False,
+            )
+        if self.options["include_dependencies"]:
+            repos += self._get_repo_dependencies(dependencies, include_beta=False)
+
+        for package in repos:
+            self._walk_releases(package)
+
         self._write_results()
 
     def _init_schema(self):
         """Initialize the structure used for schema storage."""
-        self.schema = defaultdict(
-            lambda: {"version": None, "fields": defaultdict(lambda: {"version": None})}
-        )
+        self.sobjects = defaultdict(lambda: [])
+        self.fields = defaultdict(lambda: [])
+        self.package_versions = defaultdict(lambda: [])
```

Comment:
```suggestion
        self.sobjects = defaultdict(list)
        self.fields = defaultdict(list)
        self.package_versions = defaultdict(list)
```

Reply:
One day I will internalize the "you don't need a lambda here" voice, but not just yet. Fixed.